app/src/main/java/com/example/rfid_mobilapp/NfcTagUtil.java

```diff
@@ -31,140 +31,122 @@ public static String getItemId(Intent intent, Activity activity) {
+            char[] newData = Utilities.initdata(oldData);
+            char[] newDataWithBarcode = NfcTagUtil.setBarcode(itemId, newData);
+            byte[] newDataToWrite = new String(newDataWithBarcode).getBytes(StandardCharsets.UTF_8);
 
-                byte[] cmd = getCommandWriteSingleBlock(tagId);
-                for (int i = 0; i < blocks; i++) {
-                    cmd[10] = (byte) ((offset + i) & 0x0ff);
-                    System.arraycopy(newDataToWrite, blockSize * i, cmd, 11, blockSize);
-
-                    nfcV.transceive(cmd);
-                }
-                nfcV.close();
-                Toast.makeText(activity, "Success to write to the tag. The new itemId is " + itemId, Toast.LENGTH_LONG).show();
-            } catch (IOException ioException) {
-                Toast.makeText(activity, "Failed to write to the tag", Toast.LENGTH_LONG).show();
-            }
+            writeBlocks(tagId, nfcV, activity, 0, 8, newDataToWrite);
         }
     }
 
-    public static void checkIn(Intent intent, Activity activity) {
-        if (intent != null) {
-            Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
-            NfcV nfcV = NfcV.get(tag);
+    private static void writeBlocks(byte[] tagId, NfcV nfcV, Activity activity, int offset, int blocks, byte[] newDataToWrite) {
+        try {
+            nfcV.connect();
+            byte[] cmd = getCommandWriteSingleBlock(tagId);
+            for (int i = 0; i < blocks; i++) {
+                cmd[10] = (byte) ((offset + i) & 0x0ff);
+                System.arraycopy(newDataToWrite, 4 * i, cmd, 11, 4);
 
-            try {
-                nfcV.connect();
-                byte[] tagId = tag.getId();
-                byte[] cmd = getCommand(tagId, writeAFICommand, checkInValue);
                 nfcV.transceive(cmd);
-
-                nfcV.close();
-                Toast.makeText(activity, "Success to check in.", Toast.LENGTH_LONG).show();
-            } catch (IOException ioException) {
-                Toast.makeText(activity, "Failed to check in", Toast.LENGTH_LONG).show();
             }
+            nfcV.close();
+            Toast.makeText(activity, "Success to write to the tag. The new itemId is ", Toast.LENGTH_LONG).show();
+        } catch (IOException ioException) {
+            Toast.makeText(activity, "Failed to write to the tag", Toast.LENGTH_LONG).show();
         }
     }
 
+    public static void check(Intent intent, Activity activity, String checkValue) {
 
-    public static void checkOut(Intent intent, Activity activity) {
         if (intent != null) {
             Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
             NfcV nfcV = NfcV.get(tag);
 
             try {
                 nfcV.connect();
                 byte[] tagId = tag.getId();
-                byte[] cmd = getCommand(tagId, writeAFICommand, checkOutValue);
-
+                byte[] cmd;
+                if (checkValue.equals("false")) {
```

Comment:
I think I would have prefered a proper boolean flag being passed around, instead of a String value, for this purpose at least.